### DISCUSSION TITLE
Phase 2 - Dropbox Files

### DISCUSSION THREAD

**Discussion by alvarix** [alvarix/FLogger#70] [Ideas]
@chadcrume 
**Thinking about next steps.**
Instead of displaying file contents show an edit button that sends the file contents as a prop
to addNote. 

**From our tags discussion:**
_Focus on logs that are single files with a single title containing multiple entries._

This implies that with a given file we can: 
1. add a new timestamped entry
2. select older entries to edit (?) (and add an array of edit timestamps)

Implies that instead of loading the contents of a file into a form to create edits our interface consists of:
1. an empty input to add a new entry
2. a list of past entries that are clickable to load into the input to edit

**Comment by chadcrume** [comment]
Rephrasing for discussion:

**Component: Form to add new entry.** — New entries are timestamped.

**Component: List of past entries.** — Only most recent version of each entry.

**Component: Each entry** is clickable to edit. — Editing adds a
metadata history line of text to the end of the content. And each entry has a _copy_ button that creates a new entry with new timestamp, but copy of full content including metadata history text.

Where does the dropbox auth and data file selection fit in?

**Component: DropBox Auth**

*Unauthenticated:*

Button to connect to DropBox
Flow through DropBox auth
QUESTION: Can we automatically redirect to DropBox auth?
QUESTION: Can we persist the auth long term?

*Authenticated, no file selected:*

Button to forget DropBox auth session?
List of DropBox files. — Filtered to *.json or *.flogger or something?
Each file is clickable to load data. — File data is loaded, parsed, and passed back to the App and into the **Component: List of past entries**

*Authenticated, file selected:*

Button to close DropBox file?
Button to forget DropBox auth session?
File name that's open

**Comment by chadcrume** [comment]
Data model:
```
Entry: {
    date: OurDateStringFormat,
    content: string
}
```

File format:

> 6/28/2024
> My workout today
> Curls
> Squats
> {
>     Edited on 6/30/2024
> Copied from 6/7/2024
>     Edited on 6/10/2024
>     Edited on 6/9/2024
> }
> 
> 6/14/2024
> Something else
> 
> 6/7/2024
> Something else
> 
> 6/7/2024
> My workout today second of 2!
> Curls
> Squats
> {
>     Edited on 6/10/2024
>     Edited on 6/9/2024
> }
> 
> 6/7/2024
> My workout today first of 2!
> Curls
> Squats
> 
> 5/2/2024
> Something else
> 
>

**Comment by chadcrume** [comment]
Chad did the useLoadedNotes composable and DropBoxFiles component. 
https://github.com/alvarix/FLogger/pull/74

Next:
## Alvar 
can work on revisions to the Note component:

** Component: Updates to Note.vue — for each entry in the NoteList **

* Display in **pre** element
* Editing adds a metadata history line of text to the end of the content. 
* Each entry has a _copy_ button that creates a new entry with new timestamp, but copy of full content including metadata history text.
* Using the useLoadedNotes composable to save new notes
* Handle date formatting
* Get rid of grid layout

## Chad 
can work on writing results back to file in DropBox.

**Comment by chadcrume** [comment]
Alvar made Add form toggled

Next:

### Alvar

can work on revisions to the Note component:

**_Refactor UI changing Note -> Entry_**


**_Component: Add Entry_**

- Make add entry look like existing entries 


**_Component: Add File_**

- add file can display add note inside


**_Updates to component Entrylist_**

- File opened: New Log Entry
- Edit button on each entry
- If no note opened: New File


**_Component: Updates to Note.vue (to become Entry.vue) — for each entry in the Entrylist_**

- Display in pre element
- Editing adds a metadata history line of text to the end of the content.
- Each entry has a copy button that creates a new entry with new timestamp, but copy of full content including metadata history text.
- Using the useLoadedNotes composable to save new notes
- Handle date formatting
- Get rid of grid layout


### Chad

can work on writing results back to file in DropBox.

**Comment by chadcrume** [comment]
Alvar pushed hiding AddNote if file is loaded. Next up is to Refactor changing Note -> Entry

Chad in progress with writing results back to file in DropBox.

Same tasks in queue as before.

**Comment by alvarix** [comment]
Alvar next: 

1. refactor Note to Entry
2. Put back universal add entry whether file is open or not
  - After Submit, in App.vue addNewNote() checks if file is open and continues
  - Else asks to create a new File in App.vue
  - In the future we want the option to select an existing File to add to or create a new File
 
Chad next:
- Continue abstracting the data source interface 
- Implement saving to Dropbox
